### ISSUE TITLE
Feature Request: svg support

### ISSUE THREAD

**Issue by kcmw3e** [pyglet/pyglet#192]
Hi,

I am wondering if pyglet could implement support for .svg files within the image module (or unless there's a better place for it). Unless I've missed documentation on how to do it, is this feasible? If it is and there's no one to work on it, I'd be willing to take a stab but I would need a little direction for it.

Thanks!

**Comment by benmoran56** [comment]
Hi @kcmw3e, thanks for opening the ticket. There are no plans to do this at the moment, but if it's something you're interested in working on, I'd be happy to offer you my support. I would suggest joining the Discord server https://discord.gg/QXyegWe and dropping by the "contributing" channel. 

There is already an experimental proof of concept in the repository. That can be found here:  https://github.com/pyglet/pyglet/blob/master/experimental/svg_test.py
It's incomplete, but might give some ideas. 

As in the example, it seems like it's most logical that SVGs be constructed of primitives like lines and triangles. Once that part is done, it could then be rasterized into an image, but I would leave that for last.

**Comment by matanox** [comment]
[Here's what Claude (currently) thinks about converting SVG to OpenGL](https://claude.ai/share/4bdf6e09-d1a7-48ab-9624-c18eca7fcde3). I used JetBrains Junie to bootstrap an initial attempt at it, it only worked for some SVG files which I can't share as my license to them does not allow, and for those SVG that it could convert, its line building strategy was sometimes leaving undrawn gaps in the curves that it is drawing. 

Helpful python libraries which Claude suggested for this were the following ones.

numpy>=1.19.0
svgpathtools>=1.4.1
triangle>=20200424

I guess that Animation is the main/only feature of SVG that can be left out of scope as it is clear-cut to define, but the rest of the requirements list above from Claude is not very humble.

On the other side again, it did manage to convert some of my simpler SVG files.

**Comment by matanox** [comment]
> Hi,
> 
> I am wondering if pyglet could implement support for .svg files within the image module (or unless there's a better place for it). Unless I've missed documentation on how to do it, is this feasible? If it is and there's no one to work on it, I'd be willing to take a stab but I would need a little direction for it.
> 
> Thanks!

[Actually on linux some underlying SVG library is being used by pyglet to successfully render svg, or at least it used to last time I tried](https://github.com/pyglet/pyglet/issues/642#issuecomment-1174789836). If only other platforms had been as generous to supply this capability ...

**Comment by caffeinepills** [comment]
Closing as not planned for a core feature.
We do have an example of limited SVG displaying and loading that we plan to support going forward: https://github.com/pyglet/pyglet/tree/development/contrib/svg

If someone wants to expand on this to add more SVG capabilities, they are free to do so. At the very least the basic SVG format is supported via that example.
